crates/services/txpool_v2/src/service.rs

```diff
@@ -1,234 +1,157 @@
         GasPriceProvider as GasPriceProviderTrait,
-        MemoryPool as MemoryPoolTrait,
+        P2PRequests,
+        P2PSubscriptions,
         TxPoolPersistentStorage,
         WasmChecker as WasmCheckerTrait,
     },
-    selection_algorithms::ratio_tip_gas::RatioTipGasSelection,
+    selection_algorithms::{
+        ratio_tip_gas::RatioTipGasSelection,
+        Constraints,
+    },
+    service::{
+        memory::MemoryPool,
+        p2p::P2PExt,
+        pruner::TransactionPruner,
+        subscriptions::Subscriptions,
+        verifications::Verification,
+    },
+    shared_state::SharedState,
     storage::{
         graph::{
             GraphConfig,
             GraphStorage,
         },
         Storage,
     },
-    verifications::perform_all_verifications,
+    update_sender::TxStatusChange,
+};
+use fuel_core_services::{
+    AsyncProcessor,
+    RunnableService,
+    RunnableTask,
+    ServiceRunner,
+    StateWatcher,
+    SyncProcessor,
+};
+use fuel_core_types::{
+    fuel_tx::{
+        Transaction,
+        TxId,
+        UniqueIdentifier,
+    },
+    fuel_types::{
+        BlockHeight,
+        ChainId,
+    },
+    services::{
+        block_importer::SharedImportResult,
+        p2p::{
+            GossipData,
+            GossipsubMessageInfo,
+            PeerId,
+            TransactionGossipData,
+        },
+        txpool::ArcPoolTx,
+    },
+    tai64::Tai64,
+};
+use futures::StreamExt;
+use parking_lot::RwLock;
+use std::{
+    collections::VecDeque,
+    sync::Arc,
+    time::SystemTime,
+};
+use tokio::{
+    sync::{
+        mpsc,
+        oneshot,
+        Notify,
+    },
+    time::MissedTickBehavior,
 };
 
-pub type RemovedTransactions = Vec<PoolTransaction>;
-pub type InsertionResult = Result<RemovedTransactions, Error>;
+pub(crate) mod memory;
+mod p2p;
+mod pruner;
+mod subscriptions;
+pub(crate) mod verifications;
 
-pub type TxPool<PSProvider> = Arc<
-    RwLock<
-        Pool<
-            PSProvider,
-            GraphStorage,
-            <GraphStorage as Storage>::StorageIndex,
-            BasicCollisionManager<<GraphStorage as Storage>::StorageIndex>,
-            RatioTipGasSelection<GraphStorage>,
-        >,
-    >,
+pub type TxPool = Pool<
+    GraphStorage,
+    <GraphStorage as Storage>::StorageIndex,
+    BasicCollisionManager<<GraphStorage as Storage>::StorageIndex>,
+    RatioTipGasSelection<GraphStorage>,
 >;
 
-pub struct SharedState<
-    PSProvider,
-    ConsensusParamsProvider,
-    GasPriceProvider,
-    WasmChecker,
-    MemoryPool,
-> {
-    pool: TxPool<PSProvider>,
-    persistent_storage_provider: Arc<PSProvider>,
-    current_height: Arc<RwLock<BlockHeight>>,
-    consensus_parameters_provider: Arc<ConsensusParamsProvider>,
-    gas_price_provider: Arc<GasPriceProvider>,
-    wasm_checker: Arc<WasmChecker>,
-    memory: Arc<MemoryPool>,
-    heavy_async_processor: Arc<HeavyAsyncProcessor>,
-    utxo_validation: bool,
-}
+pub type Shared<T> = Arc<RwLock<T>>;
```

Comment:
While this is a pretty nice type-alias, I don't see any point in making it public. Any other modules that would benefit from the type alias can redefine it, which would be more straight-forward to read and just as concise as importing it without the extra indirection.

```suggestion
type Shared<T> = Arc<RwLock<T>>;
```

Reply:
Optionally if this is used in a lot of places we should move it out to another module, since the alias has nothing to do with this particular service.

Reply:
I changed it to pub(crate) because it's used in the tests